src/bosh-monitor/lib/bosh/monitor/director.rb

```diff
@@ -68,7 +68,15 @@ def perform_request(method, uri, options = {})
       headers = {}
       headers['authorization'] = auth_provider.auth_header unless options.fetch(:no_login, false)
 
-      http = EM::HttpRequest.new(target_uri).send(method.to_sym, head: headers)
+      unless use_proxy?(target_uri, ENV['no_proxy'] || ENV['NO_PROXY'] || '')
+        conoptions = HttpConnectionOptions.new(target_uri, proxy: { host: '127.0.0.1', port: 25555, type: :http })
+        conoptions.https = false
+      end
+
+      http_con = EM::HttpConnection.new
```

Comment:
I intended to add some comments to this place in the code, but this is in your personal Github "organization" and I don't have privs to push, so I leave a comment here.

I would add comments that say two things:
1) This code is ugly, but it exists to add in handling for the `NO_PROXY` environment variable. The intention is for the code to check if the host is in the `NO_PROXY` list, and -if it is- create a special `HttpConnectionOptions` object that is configured so that `em-http-request` will treat the host as a plain HTTP "proxy", and just talk HTTP(S) directly to it, and replace the HttpConnectionOptions object that `em-http-request` makes for the `HttpConnection` with our specially-constructed one.
2) To future programmers: If you can find a reasonable way to call `HttpRequest.new` and add our special `HttpConnectionOptions` to the connection that it returns, rather than having code that duplicates what the `HttpRequest` initializer does, then do that. 

I guess if _you_ can verify that you can do thing #2 now, then doing that would be better than leaving a comment encouraging others to do it in the future.